source/common/matcher/map_matcher.h

```diff
@@ -27,11 +29,12 @@ class MapMatcher : public MatchTree<DataType>, Logger::Loggable<Logger::Id::matc
       return {MatchState::UnableToMatch, absl::nullopt};
     }
 
-    if (!input.data_) {
+    // Returns `on_no_match` when input data is empty. (i.e., is absl::monostate).
+    if (absl::holds_alternative<absl::monostate>(input.data_)) {
```

Comment:
This code for now assumes that there are only two types in the variant. It will break and need to be fixed once more types are added. Since this matches expects `std::string`, it perhaps should return `on_no_match_` any time the type is **not** `std:string`. In this way this code does not need to change as more types are added to the variant. What do you think.

There are other matchers similar to this down the line.